e2etest/pkg/metallb/config.go

```diff
@@ -27,6 +28,11 @@ func init() {
 
 // PeersForContainers returns the metallb config peers related to the given containers.
 func PeersForContainers(containers []*frrcontainer.FRR, ipFamily ipfamily.Family) []metallbv1beta2.BGPPeer {
+	return PeersForContainersWithSecret(containers, ipFamily, false)
+}
+
+// PeersForContainersWithSecret returns the metallb config peers related to the given containers.
+func PeersForContainersWithSecret(containers []*frrcontainer.FRR, ipFamily ipfamily.Family, useSecret bool) []metallbv1beta2.BGPPeer {
```

Comment:
I would use a vararg with a modifier function, pretty much like we do in https://github.com/metallb/metallb/blob/10a02498f259870492a2f3992803f0a7e1a4995d/e2etest/pkg/service/create.go#L20

and expose a `WithSecret` func that does the modification.